common/messaging/kafka/clientImpl.go

```diff
@@ -175,19 +175,26 @@ func convertTLSConfig(tlsConfig auth.TLS) (*tls.Config, error) {
 		return nil, nil
 	}
 
-	cert, err := tls.LoadX509KeyPair(tlsConfig.CertFile, tlsConfig.KeyFile)
-	if err != nil {
-		return nil, err
-	}
-	caCertPool := x509.NewCertPool()
-	pemData, err := ioutil.ReadFile(tlsConfig.CaFile)
-	if err != nil {
-		return nil, err
+	if tlsConfig.CertFile != "" && tlsConfig.CaFile != "" && tlsConfig.KeyFile != "" {
+		cert, err := tls.LoadX509KeyPair(tlsConfig.CertFile, tlsConfig.KeyFile)
+		if err != nil {
+			return nil, err
```

Comment:
This isn't technically correct. You can use just a caFile and it works to enable TLS. @longquanzheng 

our version looked something like this:
```
	config := tls.Config{
	if err != nil {			InsecureSkipVerify: !tlsConfig.EnableHostVerification,
		return nil, err	
	}		}
	caCertPool := x509.NewCertPool()	
	pemData, err := ioutil.ReadFile(tlsConfig.CaFile)		if tlsConfig.CaFile != "" {
	if err != nil {			caCertPool := x509.NewCertPool()
		return nil, err			pemData, err := ioutil.ReadFile(tlsConfig.CaFile)
		if err != nil {
			return nil, err
		}
		caCertPool.AppendCertsFromPEM(pemData)

		config.RootCAs = caCertPool
	}

	if tlsConfig.CertFile != "" && tlsConfig.KeyFile != "" {
		cert, err := tls.LoadX509KeyPair(tlsConfig.CertFile, tlsConfig.KeyFile)
		if err != nil {
			return nil, err
		}

		config.Certificates = []tls.Certificate{cert}
	}		}
	caCertPool.AppendCertsFromPEM(pemData)	


	return &tls.Config{		return &config, nil
```